Assets/MixedRealityToolkit/_Core/Services/MixedRealityToolkit.cs

```diff
@@ -879,7 +916,7 @@ public List<IMixedRealityService> GetActiveServices(Type type)
             if (type == null)
             {
                 Debug.LogWarning("Unable to get managers with a type of null.");
-                return new List<IMixedRealityService>();
+                return null;
```

Comment:
This was causing large performance issues with memory allocations each time a service is not found, and searching continues (sometimes indefinitely).

null is the appropriate response here to prevent these problems, and we're not decorating the methods with a `[NotNull]` attribute, so it's pretty much obvious that the returned value could potentially be null.

When requesting a service by name, it's expected to return null if the service isn't found. Why should that be any different when we're searching for services of a specific type?

Reply:
You do bring up a good point tho. This change isn't really related. I should probably add an issue and split this out into its own PR.

Reply:
>so it's pretty much obvious that the returned value could potentially be null

But also possible to have looked at the code and seen that it never returns null. (A different topic, but) it might be good to more consistently use the `TryGet` nomenclature if the method might fail to return something you can use.

>Why should that be any different when we're searching for services of a specific type?

I see it as different because an empty list still tells you that no services were found. There's not really a "default empty" service to return if no service was found, so `null` makes sense.

>I should probably add an issue and split this out into its own PR.

I'm all for that.